src/main/java/appeng/api/crafting/IPatternDetailsDecoder.java

```diff
@@ -36,5 +38,12 @@
     boolean isEncodedPattern(ItemStack stack);
 
     @Nullable
-    IPatternDetails decodePattern(ItemStack stack, Level level, boolean autoRecovery);
+    IPatternDetails decodePattern(AEItemKey what, Level level);
```

Comment:
Could be defaulted?

Reply:
Or I'd just leave it out of the Decoder itself tbh, it should only be in the registry.